src/class.cc

```diff
@@ -12,14 +12,83 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+#include "class.h"
+
 namespace libwebp {
 
-// Let's implement stuff !
-// TODO: to create an animation:
-//       - create a WebPAnimEncoderOptions  config
-//       - set the quality to something in your search space
-//       - call WebPAnimEncoderNew, then fill the animation with WebPPicture
-//         using WebPAnimEncoderAdd
-//       - finalize with WebPAnimEncoderAssemble
+Thumbnailer::Thumbnailer() {
+  WebPAnimEncoderOptionsInit(&anim_config);
+  WebPConfigInit(&config);
+}
+
+Thumbnailer::~Thumbnailer() { WebPAnimEncoderDelete(enc); }
+
+Thumbnailer::Status Thumbnailer::AddFrame(const WebPPicture& pic,
+                                          int timestamp_ms) {
+  // Verifies dimension of frames.
+  if (frames.empty()) {
+    enc = WebPAnimEncoderNew(pic.width, pic.height, &anim_config);
+    if (enc == nullptr) {
+      return kMemoryError;
+    }
+  } else {
+    if (pic.width != (frames[0].pic).width ||
+        pic.height != (frames[0].pic).height) {
+      return kImageFormatError;
+    }
+  }
+
+  FrameData new_frame = {pic, timestamp_ms};
+  frames.push_back(new_frame);
+
+  return kOk;
+}
+
+Thumbnailer::Status Thumbnailer::GenerateAnimation(WebPData* const webp_data) {
+  bool ok = true;
+
+  // Rearranges frames.
+  std::sort(frames.begin(), frames.end(),
+            [](const FrameData& A, const FrameData& B) -> bool {
+              return A.timestamp_ms < B.timestamp_ms;
+            });
+
+  // Fills the animation.
+  for (auto& frame : frames) {
+    ok = ok && WebPAnimEncoderAdd(enc, &frame.pic, frame.timestamp_ms, &config);
+  }
+
+  // Adds last frame.
```

Comment:
for a comment, please keep direct style: Add last frame.
The Adds last frame is just for a comment for a function. I know ...